german translation + load system language

olwig opened the pull request:
fix: automatically localize using system language
i18n: added german translation

nick42d commented:
Shall take a look at this when time permits, are you sure that Rust code needs to be modified? The contributor of the Swedish translation didn't raise any issue just adding a .FTL file.

olwig commented:
I tested this again with a clean german cosmic setup: cloned your latest master, added my language file to de/, ran just build-release and sudo just install, and added the applet to the panel. But it's still showing in English.

I also tried placing the file in de-DE/ and de_DE/, same result.
Only when I compile with the fallback explicitly set to de, it actually shows up in German. 

With my fix, the language is loaded automatically right away.

I checked the applet repo here:
https://github.com/pop-os/cosmic-applets
Their localization setup looks a little different. They use a localizer() function. localize() is also called before run(). The localization trigger is local to the applet.

I might be missing something, but it looks clear to me. I'm not deep into all the rust borrowing stuff yet .. .why they box the localizer. But I can just adapt their pattern to my code. Then it would match the pattern they use.

nick42d commented:
Ok, when I get a chance I'll do some testing and reading into this so that
I can provide a decent review. Unfortunately not too experienced in i18n as
I just copied the setup from the cosmic applet template at the time.

Happy to merge just the .FTL file quickly, if you split this PR into 2 in
that way.

Cheers Nick

On Tue, 13 May 2025, 04:02 olwig, ***@***.***> wrote:

> *olwig* left a comment (nick42d/cosmic-applet-arch#26)
> <https://github.com/nick42d/cosmic-applet-arch/pull/26#issuecomment-2873827991>
>
> I tested this again with a clean german cosmic setup: cloned your latest
> master, added my language file to de/, ran just build-release and sudo just
> install, and added the applet to the panel. But it's still showing in
> English.
>
> I also tried placing the file in de-DE/ and de_DE/, same result.
> Only when I compile with the fallback explicitly set to de, it actually
> shows up in German.
>
> With my fix, the language is loaded automatically right away.
>
> I checked the applet repo here:
> https://github.com/pop-os/cosmic-applets
> Their localization setup looks a little different. They use a localizer()
> function. localize() is also called before run(). The localization trigger
> is local to the applet.
>
> I might be missing something, but it looks clear to me. I'm not deep into
> all the rust borrowing stuff yet .. .why they box the localizer. But I can
> just adapt their pattern to my code. Then it would match the pattern they
> use.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nick42d/cosmic-applet-arch/pull/26#issuecomment-2873827991>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A727HYZMHO5D5OZ5XQBP3S326D45FAVCNFSM6AAAAAB44GJMZOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNZTHAZDOOJZGE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

olwig commented:
I reset the branch to the point where I had just added the translation file. The other changes are in the [fix-auto-localize](https://github.com/olwig/cosmic-applet-arch/tree/fix-auto-localize) branch in my fork, in case you still need them.

nick42d commented:
Thanks very much! I shall merge the new translation, please raise a new PR for the fix-auto-localize branch if you'd like the PR cred for the localization fix too else I shall manually bring it over when I get a chance.
